Source/Core/DolphinQt/Achievements/AchievementProgressWidget.cpp

```diff
@@ -49,6 +51,45 @@ AchievementProgressWidget::AchievementProgressWidget(QWidget* parent) : QWidget(
 QGroupBox*
 AchievementProgressWidget::CreateAchievementBox(const rc_api_achievement_definition_t* achievement)
 {
+  QLabel* a_badge = new QLabel();
+  const auto unlock_status = AchievementManager::GetInstance()->GetUnlockStatus(achievement->id);
+  auto badge = const_cast<AchievementManager::BadgeStatus*>(&unlock_status.locked_badge);
+  std::string_view color = AchievementManager::GRAY;
+  if (unlock_status.remote_unlock_status == AchievementManager::UnlockStatus::UnlockType::HARDCORE)
+  {
+    badge = const_cast<AchievementManager::BadgeStatus*>(&unlock_status.unlocked_badge);
+    color = AchievementManager::GOLD;
+  }
+  else if (hardcore_mode_enabled || unlock_status.session_unlock_count > 1)
+  {
+    badge = const_cast<AchievementManager::BadgeStatus*>(&unlock_status.unlocked_badge);
+    color = AchievementManager::GOLD;
+  }
+  else if (unlock_status.remote_unlock_status ==
+           AchievementManager::UnlockStatus::UnlockType::SOFTCORE)
+  {
+    badge = const_cast<AchievementManager::BadgeStatus*>(&unlock_status.unlocked_badge);
+    color = AchievementManager::BLUE;
+  }
+  else if (unlock_status.session_unlock_count > 1)
+  {
+    badge = const_cast<AchievementManager::BadgeStatus*>(&unlock_status.unlocked_badge);
+    color = AchievementManager::BLUE;
+  }
+  if (Config::Get(Config::RA_BADGES_ENABLED) && badge->name != "")
+  {
+    QImage i_badge{};
+    if (i_badge.loadFromData(&badge->badge.front(), (int)badge->badge.size()))
+    {
+      a_badge->setPixmap(QPixmap::fromImage(i_badge).scaled(64, 64, Qt::KeepAspectRatio,
+                                                            Qt::SmoothTransformation));
+      a_badge->adjustSize();
+      a_badge->setStyleSheet(
+          QStringLiteral("border: 4px solid %1").arg(QString::fromStdString(std::string(color))));
+      a_badge->setVisible(true);
```

Comment:
This is the source of your mysterious per-badge popups. Just remove this line, it's unnecessary and only confuses Qt about your intent. There's probably a similar line somewhere else that causes the popup when you open the achievement window.

Reply:
Curious what exactly makes this unnecessary, is it just that this should default to visible unless I specify otherwise, or does it have something to do with the stylesheet?

Reply:
Someone with more Qt knowledge than me would have to answer this properly, but I think it has to do with how Qt hierarchies work; if a widget has a parent, its visibility state controls whether it is visible within that parent, but if it doesn't have a parent setting it visible opens a new top-level window containing the widget. Something like that.

Reply:
OH. Okay, that tracks. I'll check where else I'm setting that to account for it.

Reply:
Here's a fix for all of those instances I could find _outside_ of the badges, fyi. https://github.com/dolphin-emu/dolphin/pull/12205